### PR TITLE
Improve diagnose.py, adding build features info and binary library path.

### PR DESCRIPTION
**Pull request by larroy**
See title.

This is the script used to fill issues, so adding more info will help.

Refined output:

[diagnose.txt](https://github.com/apache/incubator-mxnet/files/3375177/diagnose.txt)

### CODE REVIEW

**Comment by wkcn** [comment]
Thank you for the improvement!

I have a suggestion : )
Could we only print the relative path?
Some users don't like the printing of the absolute path with their privacy.

File: tools/diagnose.py [hunk 1]
```diff
@@ -105,13 +110,20 @@ def check_mxnet():
...
+        if os.path.exists(commit_hash):
+            with open(commit_hash, 'r') as f:
+                ch = f.read().strip()
+                print('Commit Hash   :', ch)
+        else:
+            print('Commit hash file "{}" not found. Not installed from pre-built package or built from source.'.format(commit_hash))
+        print('Library      :', mxnet.libinfo.find_lib_path())
+        try:
+            print('Build features:')
+            print(get_build_features_str())
```
Comment (apeforest) [inline tools/diagnose.py]: This will fail if user use this script to diagnose an mxnet script before 1.5.0 release.

**Comment by apeforest** [comment]
One minor comment: the ✖ etc. symbols were not recognized when I import the txt file into MS office or Google Doc. Can we replace these unicode symbols by simple ascii, like Y/N?

**Comment by larroy** [comment]
I think they are valid unicode symbols, what problem are you experiencing with MS Office? I think that is out of scope of this PR as it requires changes to the Python code that prints the feature, feel free to send a different PR for that.

**Comment by larroy** [comment]
@apeforest I don't have any problems with Google docs or Office with those unicode characters.
Screen Shot 2019-07-13 at 7 56 35 PM
Screen Shot 2019-07-13 at 7 57 49 PM

**Pull request merged**